include/structopt/parser.hpp

```diff
@@ -57,36 +66,37 @@ struct parser {
     return result;
   }
 
-  bool is_kebab_case(const std::string &next, const std::string &field_name) {
-    bool result = false;
+  bool is_kebab_case(std::string_view next, std::string_view field_name) {
     auto maybe_kebab_case = next;
     if (maybe_kebab_case.size() > 1 && maybe_kebab_case[0] == '-') {
       // remove first dash
-      maybe_kebab_case.erase(0, 1);
+      maybe_kebab_case.remove_prefix(1);
       if (maybe_kebab_case[0] == '-') {
         // there is a second leading dash
         // remove it
-        maybe_kebab_case.erase(0, 1);
+        maybe_kebab_case.remove_prefix(1);
       }
-      std::replace(maybe_kebab_case.begin(), maybe_kebab_case.end(), '-', '_');
-      if (maybe_kebab_case == field_name) {
-        result = true;
+      if (equal_strings_replace_hyphens(maybe_kebab_case, field_name)) {
+        return true;
       }
     }
-    return result;
+    return false;
   }
 
-  bool is_optional_field(const std::string &next, const std::string &field_name) {
-    bool result = false;
-    if (next == "-" + field_name || next == "--" + field_name ||
-        next == "-" + std::string(1, field_name[0]) || is_kebab_case(next, field_name)) {
-      // okay `next` matches _a_ field name (which is an optional field)
-      result = true;
+  bool is_optional_field(std::string_view next, std::string_view field_name) {
+    if (next.rfind("-", 0) == 0 && next.substr(1) == field_name) {
+      return true;
     }
-    return result;
+    if (next.rfind("--", 0) == 0 && next.substr(2) == field_name) {
+      return true;
+    }
+    if (next == std::data({'-', field_name[0]})) {
```

Comment:
The \0 must be added here as third character, because string view constructor (at implicit cast for ==) checks the const char* until that end char. If you don't want, you need to call explicitly the string_view constructor with size `std::string_view(std::data(...), 2)`

Reply:
Good catch - fixed